js_modules/dagster-ui/packages/ui-core/src/runs/CapturedLogPanel.tsx

```diff
@@ -33,28 +35,85 @@ export const CapturedOrExternalLogPanel = React.memo(
       (props.visibleIOType === 'stdout'
         ? logCaptureInfo.externalStdoutUrl
         : logCaptureInfo.externalStderrUrl);
-    if (externalUrl) {
+    const uriOrPath =
+      logCaptureInfo &&
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutUriOrPath
+        : logCaptureInfo.stderrUriOrPath);
+    const shellCmd =
+      logCaptureInfo &&
+      (props.visibleIOType === 'stdout'
+        ? logCaptureInfo.stdoutShellCmd
+        : logCaptureInfo.stderrShellCmd);
+
+    const copy = useCopyToClipboard();
+    const onClickFn = async (key: string, value: string | undefined) => {
+      if (!value) {
+        return;
+      }
+      copy(value);
+      await showSharedToaster({
+        intent: 'success',
+        icon: 'done',
+        message: `${key} string copied!`,
+      });
+    };
+    const onClickExternalUri = async () => onClickFn('log artefact URI', uriOrPath);
+    const onClickShellCmd = async () => onClickFn('shell command', shellCmd);
+
+    if (externalUrl || uriOrPath || shellCmd) {
       return (
         <Box
-          flex={{direction: 'row', alignItems: 'center', justifyContent: 'center', gap: 1}}
+          flex={{direction: 'column', alignItems: 'center', justifyContent: 'center', gap: 1}}
           background={Colors.tooltipBackground()}
           style={{color: Colors.tooltipText(), flex: 1, minHeight: 0}}
         >
```

Comment:
This could have a `Box` to provide padding around the `Table` (something like `padding={20}` perhaps), but I think it should not have any special `background` or `style` applied to it. The default background and text color should be sufficient here.